src/msg/producer/writer/consumer_writer.go

```diff
@@ -448,25 +449,47 @@ func (w *consumerWriterImpl) reset(opts resetOptions) {
 }
 
 func (w *consumerWriterImpl) connectNoRetry(addr string) (io.ReadWriteCloser, error) {
```

Comment:
Confirmed with @vdarulis offline: we always want to set `WriteTimeout` regardless of whether keep alive is set; previous behavior (don't set WriteTimeout if KeepAlive is unset) was probably a bug that we just never hit because keep alive is always set.

That is, we should *always* set the write timeout. cc @cw9